Added 500.30 Key Vault note - a common issue (#16597)
aspnet/AspNetCore.Docs: aspnetcore/test/troubleshoot-azure-iis.md

The [ASP.NET Core Module](xref:host-and-deploy/aspnet-core-module) attempts to start the .NET Core CLR in-process, but it fails to start. The cause of a process startup failure can usually be determined from entries in the Application Event Log and the ASP.NET Core Module stdout log.

Common failure conditions:

* The app is misconfigured due to targeting a version of the ASP.NET Core shared framework that isn't present. Check which versions of the ASP.NET Core shared framework are installed on the target machine.

> * Using Azure Key Vault, lack of permissions to the Key Vault. Check the access policies in the targeted Key Vault to ensure the website user has access to read secrets.

Reviewer: ```suggestion
* Using Azure Key Vault, lack of permissions to the Key Vault. Check the access policies in the targeted Key Vault to ensure that **Get** and **List** secret permissions are granted.
```

... something like that. I don't recommend that it say "website user" (e.g., Managed Identities). The exact permissions are "Get" and "List," and those are "secrets" permissions settings.

A common failure condition is the app is misconfigured due to targeting a version of the ASP.NET Core shared framework that isn't present. Check which versions of the ASP.NET Core shared framework are installed on the target machine.

> If using Azure Key Vault, another common failure condition is lack of permissions to the Key Vault. Check the access policies in the targeted Key Vault to ensure the website user has access to read secrets. 

Reviewer 2: A common failure condition with Azure Key Vault is lack of permissions to the Key Vault. Check the access policies in the targeted Key Vault to ensure the website user has access to read secrets.
